src/main/scala/beam/utils/map/R5MapStatsCalculator.scala

```diff
@@ -1,24 +1,170 @@
+      .toSeq
+      .sortBy(_._2)
+      .reverse
+
+    println(frequentTagsToCnt.map { case (tag, cnt) => s"$tag [$cnt]" }.mkString("  "))
+  }
+
+  private def printOSMTagInfo(osm: OSM): Unit = {
+    val highwayType2Way: Map[String, Seq[(String, lang.Long, Way)]] = osm.ways.asScala
+      .map { way =>
+        val tags = way._2.tags
+        val highwayTag = tags.asScala.collectFirst { case tag if tag.key == "highway" => tag.value }
+        val highwayType = highwayTag.getOrElse("unknown")
+        (highwayType, way._1, way._2)
+      }
+      .toSeq
+      .groupBy { case (highwayType, _, _) => highwayType }
+
+    case class TagInfo(tagName: String, hasTag: Int, doesNotHaveATag: Int, values: Seq[String])
+    def toTagInfo(ways: Iterable[(String, lang.Long, Way)], tagToCollect: String): TagInfo = {
+      val tagVals = ways.map {
+        case (_, _, way) =>
+          if (way.hasTag(tagToCollect)) Some(way.getTag(tagToCollect))
+          else None
+      }
+      val doesNotHaveATag = tagVals.count(tv => tv.isEmpty)
+      val doesHaveATag = ways.size - doesNotHaveATag
+
+      TagInfo(tagToCollect, doesHaveATag, doesNotHaveATag, tagVals.flatten.toSeq)
+    }
+
+    val allTags = Set("lanes", "maxspeed", "capacity")
+
+    println("The number of tags present for each way, grouped by highway type.")
+    val highwayType2TagInfo = highwayType2Way.map {
+      case (highwayType, ways) =>
+        val tagInfos = allTags.map { tagToAnalyze =>
+          toTagInfo(ways, tagToAnalyze)
+        }
+        highwayType -> tagInfos
+    }
+
+    highwayType2TagInfo.foreach {
+      case (highwayType, tagInfos) =>
+        val numberOfWays = tagInfos.head.hasTag + tagInfos.head.doesNotHaveATag
+        println(s"\t$highwayType: $numberOfWays")
+
+        def inPercentage(value: Int, fullValue: Int): Int = value * 100 / fullValue
+        val tagInfosStr = tagInfos.map(ti => s"${ti.tagName} (${inPercentage(ti.hasTag, numberOfWays)})").mkString(" ")
+
+        println(s"\t\t$tagInfosStr")
+    }
+    println("The configuration text based on tags values:")
+
+    val ms2mph = 2.237
+
+    highwayType2TagInfo.foreach {
+      case (highwayType, tagInfos) =>
+        val tag2Into = tagInfos.map(ti => ti.tagName -> ti).toMap
+
+        val maybeSpeedIn_mph = tag2Into.get("maxspeed") match {
+          case Some(speedTagInfo) =>
+            // expected values ~ "15 mph"
+            val sumAndLen = speedTagInfo.values
+              .map(v => v.split(" ")(0).toFloat)
+              .foldLeft((0.0, 0)) { case ((accum, size), v) => (accum + v, size + 1) }
+
+            if (sumAndLen._2 > 0) {
+              Some((sumAndLen._1 / sumAndLen._2, speedTagInfo.values))
+            } else {
+              None
+            }
+          case None => None
+        }
+
+        val maybeLanes = tag2Into.get("lanes") match {
+          case Some(lanesTagInfo) =>
+            val sumAndLen = lanesTagInfo.values
```

Comment:
This can be put into a def and called for the lanes and tag - as it is the same code...barring a variable name